src/helpers/queries.js

```diff
@@ -151,3 +151,38 @@ export const buildContactQueryUpdateGreaterThan = date => ({
     .indexFields(['_id'])
     .limitBy(1000)
 })
+
+export const buildContactQueryByEmailAdressOrPhoneNumber = (
+  addresses,
+  numbers
+) => {
+  return {
+    definition: Q(DOCTYPE_CONTACTS)
+      .where({
+        _id: {
+          $gt: null
+        },
+        $or: [
+          {
+            email: {
+              $elemMatch: {
+                address: {
+                  $in: addresses
+                }
+              }
+            }
+          },
+          {
+            phone: {
+              $elemMatch: {
+                number: {
+                  $in: numbers
+                }
+              }
+            }
+          }
+        ]
+      })
+      .indexFields(['_id'])
+  }
```

Comment:
Do we have insights about how this improves perfs? Some before/after measurements?

Reply:
For me, this commit is not a refacto: it is a feat that aims to improve performances. The "why" should be explained in the commit, as well as how it actually improves perfs ;)

Reply:
After some quick benchmark, this doesn't improve the perfomance sinificantly :(

Reply:
Nouvelle méthode : 69339 ms


7 733 ms → 8s de gagner (10% d’amélioration)

Reply:
69s to import 250 docs, this is awfully slow :cry: 
I think there is some investigation to make to understand why this takes so much time... 
But this is probably out of scope of this PR, well done for improving it ;)  
Could you indicate those numbers in the related commit?